CONTRIBUTING.md

```diff
@@ -77,6 +77,80 @@ README for more details.
 - Once a pull request is good to go, the code reviewer is the one who merges
   it, then deletes the branch.
 
+## Controller design
+
+To maintain consistency throughout the project, we've come up with the following
+design rules for our Rails controllers:
+
+### RESTful controllers
+
+* Only use CRUD methods in controllers.
+
+* Prefer adding a new controller with one of the CRUD methods over creating a
+  custom method in an existing controller. For example, if your app allows a
+  user to update their email and their password on two different pages, instead of
+  using a single controller with methods called `update_email` and
+  `update_password`, create two controllers and name the methods `update`, i.e.
+  `EmailsController#update` and `PasswordsController#update`. See
+  http://jeromedalbert.com/how-dhh-organizes-his-rails-controllers/ for more about
+  this design pattern.
+
+### Lean controllers
+* Keep as much business logic out of controllers.
+
+* Use specialized classes to handle the operations
+  * These will be Form Objects for the most part, since
+  most of what the app does is process user input via a form submission, or
+  clicking a link in email that contains a token.
+
+* Form Object rules:
+  - Must have a single public method called `submit` that returns a [FormResponse] object.
```

Comment:
I would replace all the `Must` with `Should`. Things won't break if we don't, but they could lead to confusion. `Should` in the RFCs says that we can violate it if we have a very good reason for doing so, if we understand the consequences, and we accept those consequences. It's stronger than `May`, which is just a suggestion.

One thing we need to guard against is spreading everything out so thin that we are back in spaghetti code. A developer needs to be able to hold the entirety of a problem in their mind when working on it. The more running through code that they have to do to find all the important pieces, the more difficult that becomes.

There's not a simple rule that fixes all of this, so we have to be able to balance the complexity of spreading everything out against the complexity of keeping related pieces of code together.